### PR TITLE
[order_utils.py] generate_order_hash_hex()

### PR DESCRIPTION
**Pull request by feuGeneA**
## Testing instructions

`pip install -e .[dev]`
`./setup.py test`

Also integrated with CI.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist:

*   [x] Prefix PR title with `[WIP]` if necessary.
*   [x] Prefix PR title with bracketed package name(s) corresponding to the changed package(s). For example: `[sol-cov] Fixed bug`.
*   [x] Add tests to cover changes as needed.
*   [x] Update documentation as needed.
*   [x] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/utils/test/sign_typed_data_utils_test.ts [hunk 1]
```diff
@@ -136,5 +136,28 @@ describe('signTypedDataUtils', () => {
...
+                    senderAddress: '0x0000000000000000000000000000000000000000',
+                    salt: 0,
+                    makerAssetData: '0x0000000000000000000000000000000000000000',
+                    takerAssetData: '0x0000000000000000000000000000000000000000',
+                    exchangeAddress: '0x0000000000000000000000000000000000000000',
+                },
+            };
+            const hash = signTypedDataUtils.generateTypedDataHash(uninitializedOrder).toString('hex');
+            const hashHex = `0x${hash}`;
+            expect(hashHex).to.be.eq('0xfaa49b35faeb9197e9c3ba7a52075e6dad19739549f153b77dfcf59408a4b422');
```
Comment (feuGeneA) [inline packages/utils/test/sign_typed_data_utils_test.ts]: Note that this expected value also serves as the expected value in `python-packages/order_utils/test/test_generate_order_hash_hex.py`.

File: python-packages/order_utils/src/zero_ex/order_utils/__init__.py [hunk 2]
```diff
@@ -9,3 +9,132 @@
...
+    ...         'maker_fee': 0,
+    ...         'taker_fee': 0,
+    ...         'expiration_time_seconds': 12345,
+    ...         'salt': 12345,
+    ...         'maker_asset_data': "0000000000000000000000000000000000000000",
+    ...         'taker_asset_data': "0000000000000000000000000000000000000000",
+    ...         'exchange_address': "0x0000000000000000000000000000000000000000",
+    ...     },
+    ... )
+    '55eaa6ec02f3224d30873577e9ddd069a288c16d6fb407210eecbc501fa76692'
```
Comment (feuGeneA) [inline python-packages/order_utils/src/zero_ex/order_utils/__init__.py]: Note that this expected value comes from the existing test case in `packages/utils/test/sign_typed_data_utils_test.ts`.

File: python-packages/order_utils/test/test_generate_order_hash_hex.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+"""Test zero_ex.order_utils.get_order_hash_hex()."""
+
+from zero_ex.order_utils import generate_order_hash_hex, make_empty_order
+
+
+def test_get_order_hash_hex__empty_order():
+    """Test the hashing of an uninitialized order."""
+    expected_hash_hex = (
+        "faa49b35faeb9197e9c3ba7a52075e6dad19739549f153b77dfcf59408a4b422"
```
Comment (feuGeneA) [inline python-packages/order_utils/test/test_generate_order_hash_hex.py]: Here is that usage of the expected value from the new test in `packages/utils/test/sign_typed_data_utils_test.ts`.

File: python-packages/order_utils/src/zero_ex/order_utils/__init__.py [hunk 4]
```diff
@@ -9,3 +9,145 @@
...
+
+from eth_utils import is_address, keccak, to_checksum_address, to_bytes
+from web3 import Web3
+from web3.utils import datatypes
+import web3.exceptions
+
+
+EXCHANGE_ABI = json.loads(
+    resource_string("zero_ex.contract_artifacts", "artifacts/Exchange.json")
+)["compilerOutput"]["abi"]
```
Comment (fabioberger) [inline python-packages/order_utils/src/zero_ex/order_utils/__init__.py]: Do we need to load this in every file that needs it? Or can we load it somewhere once?

File: python-packages/order_utils/src/zero_ex/order_utils/__init__.py [hunk 5]
```diff
@@ -9,3 +9,145 @@
...
+        + b"uint256 makerAssetAmount,"
+        + b"uint256 takerAssetAmount,"
+        + b"uint256 makerFee,"
+        + b"uint256 takerFee,"
+        + b"uint256 expirationTimeSeconds,"
+        + b"uint256 salt,"
+        + b"bytes makerAssetData,"
+        + b"bytes takerAssetData"
+        + b")"
+    )
```
Comment (fabioberger) [inline python-packages/order_utils/src/zero_ex/order_utils/__init__.py]: This is static, can't me move this outside of this method?

File: python-packages/order_utils/src/zero_ex/order_utils/__init__.py [hunk 6]
```diff
@@ -9,3 +9,145 @@
...
+    "1": "0x4f833a24e1f95d70f028921e27040ca56e09ab0b",
+    "3": "0x4530c0483a1633c7a1c97d2c53721caff2caaaaf",
+    "42": "0x35dd2932454449b14cee11a94d3674a936d5d7b2",
+    "50": "0x48bacb9266a570d521063ef5dd96e61686dbe788",
+}
+
+NULL_ADDRESS = "0x0000000000000000000000000000000000000000"
+
+
+class Order(TypedDict):  # pylint: disable=too-many-instance-attributes
```
Comment (fabioberger) [inline python-packages/order_utils/src/zero_ex/order_utils/__init__.py]: Is this `TypedDict` kosher?

Comment (feuGeneA) [reply python-packages/order_utils/src/zero_ex/order_utils/__init__.py]: Talked offline and resolved all concerns.

File: python-packages/order_utils/src/zero_ex/order_utils/__init__.py [hunk 7]
```diff
@@ -9,3 +9,157 @@
...
+    ...         'expiration_time_seconds': 12345,
+    ...         'salt': 12345,
+    ...         'maker_asset_data': "0000000000000000000000000000000000000000",
+    ...         'taker_asset_data': "0000000000000000000000000000000000000000",
+    ...     },
+    ...     exchange_address="0x0000000000000000000000000000000000000000",
+    ... )
+    '55eaa6ec02f3224d30873577e9ddd069a288c16d6fb407210eecbc501fa76692'
+    """  # noqa: E501 (line too long)
+    # TODO: use JSON schema validation to validate order. pylint: disable=fixme
```
Comment (fabioberger) [inline python-packages/order_utils/src/zero_ex/order_utils/__init__.py]: Will you do this in a subsequent PR?

Comment (feuGeneA) [reply python-packages/order_utils/src/zero_ex/order_utils/__init__.py]: Yes

**Pull request merged**